casper/src/main/rholang/MakePoS.rho

```diff
@@ -61,6 +62,57 @@ contract @"MakePoS"(@purse, @minimumBond, @maximumBond, @initBonds, return) = {
       }
     } |
     
+    contract feeDistribution(@n, @joiningFee, return) = {
+      new rangeCh, feeDistFunc, feeDistCh, sum, nCh, totalCh in {
+        contract sum(@x, @y, return) = { return!(x + y) } |
+        contract feeDistFunc(@k, return) = { return!((2 * joiningFee * (n + 1 - k)) / (n * (n + 1))) } |
+        @("ListOps", "range")!(1, n + 1, *rangeCh) |
+        for(@range <- rangeCh) {
+          @("ListOps", "map")!(range, *feeDistFunc, *feeDistCh) |
+          for(@feeDist <- feeDistCh) {
+            @("ListOps", "fold")!(feeDist, 0, *sum, *totalCh) |
+            for(@total <- totalCh) {
+              return!(
+                //give remainder (non-zero from rounding) to first validator
+                [feeDist.nth(0) + joiningFee - total] ++
+                  feeDist.slice(1, n)
+              )
+            }
+          }
+        }
+      }
+    } |
+
+    contract updateBonds(@publicKey, @sigVerify, @bondAmount, @rewardsForwarder, return) = {
```

Comment:
The contract name is `updateBonds` but if I am reading this correctly, [in lines 103-113](https://github.com/rchain/rchain/pull/1572/files#diff-b17a6d3ec7df93e6f6e6f8fecddabb0dR103) it is bonding new validator ?

Reply:
Is there a guard that `bondsAmount + joiningFee > 0` ? I don't see when `validateBondsAmount` contract is being called.

Reply:
Thanks for the explanation.